### PR TITLE
Recipe execution with dsl

### PR DESCRIPTION
**Pull request by prakashmrec**
Added a class RecipeExecution which can be used in DSL. It can be provided with a Recipe as a closure for local or remote execution.
Alos, refactored the DSL code a bit to hide unnecessary details from end user.

### CODE REVIEW

File: readyapi4j-groovy-dsl/src/test/groovy/com/smartbear/readyapi4j/dsl/execution/RecipeExecutionTest.groovy [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+import static com.smartbear.readyapi4j.dsl.execution.RecipeExecution.executeRecipe
+import static com.smartbear.readyapi4j.dsl.execution.RecipeExecution.executeRecipeLocally
+
+class RecipeExecutionTest extends GroovyTestCase {
+
+    void testThrowsExceptionWhenLocalRecipeExecutorNotInClasspath() throws Exception {
+        shouldFail(IllegalStateException.class, {
+            executeRecipeLocally {
+                get 'http://google.com'
```
Comment (spindelmanne) [inline readyapi4j-groovy-dsl/src/test/groovy/com/smartbear/readyapi4j/dsl/execution/RecipeExecutionTest.groovy]: This will make the test slow and possibly flaky if the internet connection is bad. I suggest that you replace it with a groovy script test ste

File: readyapi4j-groovy-dsl/src/test/groovy/com/smartbear/readyapi4j/dsl/execution/RecipeExecutionTest.groovy [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+            }
+        })
+    }
+
+    void testRecipeExecutionWithRecipeExecutor() {
+        RecipeExecutor recipeExecutor = new RecipeExecutorAdaptor()
+        Execution execution = executeRecipe recipeExecutor, {
+            get 'http://google.com'
```
Comment (spindelmanne) [inline readyapi4j-groovy-dsl/src/test/groovy/com/smartbear/readyapi4j/dsl/execution/RecipeExecutionTest.groovy]: Use groovy script step - see above

File: readyapi4j-groovy-dsl/src/test/groovy/com/smartbear/readyapi4j/dsl/execution/RecipeExecutionTest.groovy [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+import com.smartbear.readyapi4j.execution.RecipeExecutor
+
+import static com.smartbear.readyapi4j.dsl.execution.RecipeExecution.executeRecipe
+import static com.smartbear.readyapi4j.dsl.execution.RecipeExecution.executeRecipeLocally
+
+class RecipeExecutionTest extends GroovyTestCase {
+
+    void testThrowsExceptionWhenLocalRecipeExecutorNotInClasspath() throws Exception {
+        shouldFail(IllegalStateException.class, {
+            executeRecipeLocally {
```
Comment (spindelmanne) [inline readyapi4j-groovy-dsl/src/test/groovy/com/smartbear/readyapi4j/dsl/execution/RecipeExecutionTest.groovy]: I think we should shorten this to executeRecipe

**Comment by prakashmrec** [comment]
Thanks, fixed the review comments.
